ui/src/app/shared/get_workflow_params.test.ts

```diff
@@ -0,0 +1,32 @@
+import {createBrowserHistory, createPath} from 'history';
+import {getWorkflowParametersFromQuery} from './get_workflow_params';
+
+describe('get_workflow_params', () => {
+    it('should return an empty object when there are no query parameters', () => {
+        const history = createBrowserHistory();
+        const result = getWorkflowParametersFromQuery(history);
+        expect(result).toEqual({});
+    });
+
+    it('should return the parameters provided in the URL', () => {
+        const history = createBrowserHistory();
+        const path = createPath({pathname: '/workflows', search: '?parameters[key1]=value1&parameters[key2]=value2'});
+        history.location.search = path;
```

Comment:
This test is failing because `window.location.search` should only contain the query string, not the path:
```suggestion
        history.location.search = '?parameters[key1]=value1&parameters[key2]=value2';
```
Also, if you haven't already, you should install the Jest extension in the DevContainer so you can debug the tests and set breakpoints:
<img width="1315" alt="Screenshot 2024-10-17 at 11 25 15 AM" src="https://github.com/user-attachments/assets/9db131c8-0d12-4ac8-8ef9-ad8a903598b6">